mods/cnc/maps/scb30ea/scb30ea.lua

```diff
@@ -0,0 +1,376 @@
+								Utils.Do(team, function(a)
+									if not a.IsDead then
+										a.Stance = "AttackAnything"
+										IdleHunt(a)
+									end
+								end)
+							end)
+							return
+						end
+					else
+						i = i + 1
+					end
+					Trigger.AfterDelay(20, function()
+						MoveAsGroup(team, path, i, loop)
+					end)
+				end
+			end)
+		end
+	end)
+end
+
+
+IonCannonOnline = false
+ICShotsCount = 0
+FireIonCannon = function(timer)
+	if IonCannonOnline then
+		local ii = Utils.RandomInteger(1, 4)
+		local targets = { }
+		if ii < 3 then
+			targets = Nod.GetActorsByTypes(IonCannonTargets[ii])
+		else
+			targets = Nod.GetGroundAttackers()
+		end
+		if #targets > 0 then
+			local rand = Utils.RandomInteger(1, #targets + 1)
+			if not targets[rand].IsDead then
+				GDIAdvComCenter.ActivateIonCannon(targets[rand].Location)
+				if ICShotsCount < 2 then
+					ICShotsCount = ICShotsCount + 1
+					if ICShotsCount > 1 then
+						Nod.MarkFailedObjective(DestroyIonCannon)
+						Trigger.ClearAll(GDIAdvComCenter)
+					end
+				end
+				Trigger.AfterDelay(DateTime.Seconds(timer), function() FireIonCannon(timer) end)
+				return
+			end
+		end
+		Trigger.AfterDelay(DateTime.Seconds(1), function() FireIonCannon(timer) end)
+	end
+end
+
+
+CheckObjectives = function()
+	if GDI.HasNoRequiredUnits() then Nod.MarkCompletedObjective(EliminateAllGDI) end
+	if Nod.HasNoRequiredUnits() then Nod.MarkFailedObjective(EliminateAllGDI) end
+	Trigger.AfterDelay(25, CheckObjectives)
+end
+
+
+WorldLoaded = function()
+	Nod = Player.GetPlayer("Nod")
+	GDI = Player.GetPlayer("GDI")
+	Camera.Position = PlayerStart.CenterPosition
+	Flare = Actor.Create("flare", true, { Owner = Nod, Location = DefaultFlareLocation.Location })
+	Trigger.AfterDelay(DateTime.Minutes(1), function()
+		Flare.Destroy()
+	end)
+	InitObjectives(Nod)
+	EliminateAllGDI = AddPrimaryObjective(Nod, "eliminate-gdi-forces")
+	FindAllCivMoney = AddSecondaryObjective(Nod, "take-civilians-money-crates")
+	CheckObjectives()
+	InitAi()
+
+
+	Trigger.OnEnteredFootprint(WelcomeTeamCellTrigger, function(a, id)
+		if a.Owner == Nod then
+			Utils.Do(WelcomeTeam, function(a)
+				if not a.IsDead then
+					a.AttackMove(PlayerStart.Location) IdleHunt(a)
+				end
+			end)
+			Trigger.RemoveFootprintTrigger(id)
+		end
+	end)
+
+
+	Trigger.OnAllKilled( Civilians, function()
+		local cargo = Reinforcements.ReinforceWithTransport(GDI, "tran", ReiforcementsEngineers, { CPos.New(0,32), waypoint10.Location }, { CPos.New(0,32) })[2]
+		Utils.Do(cargo, function(engs)
+			if engs.Type == "e6" then
+				Trigger.OnIdle(engs, CaptureStructures)
+			end
+		end)
+	end)
+
+
+	Utils.Do( CiviliansBuildings, function(b)
+		Trigger.OnKilled(b, function() CheckVillageDestruction() end)
+	end)
+
+
+	Utils.Do( OilDerricks, function(actor)
+		Trigger.OnKilledOrCaptured( actor, function()
+			OilDerricksAirstrike()
+		end)
+	end)
+
+
+	Utils.Do( RunForHelpCivs, function(actor)
+		Trigger.OnDiscovered( actor, function()
```

Comment:
```suggestion
	Utils.Do(RunForHelpCivs, function(actor)
		Trigger.OnDiscovered(actor, function()
```

and the plenks in between.